cmd/install/install.go

```diff
@@ -54,6 +55,7 @@ func NewCommand() *cobra.Command {
 	cmd.Flags().StringVar(&opts.HyperShiftImage, "hypershift-image", version.HyperShiftImage, "The HyperShift image to deploy")
 	cmd.Flags().BoolVar(&opts.Development, "development", false, "Enable tweaks to facilitate local development")
 	cmd.Flags().BoolVar(&opts.Render, "render", false, "Render output as YAML to stdout instead of applying")
+	cmd.Flags().BoolVar(&opts.ExcludeEtcdManifests, "exclude-etcd", false, "Leave out etcd manifests")
```

Comment:
I think this is requirement from an IBM Cloud perspective. We need to have an automated way to generate all payload without including the etcd crds. 

If there's another way we would like that automation though happy to refactor

Reply:
Revisiting this, I see what you mean and now I think I see some unexpected implications of the API change. Before, it was assumed etcd is always managed, and thus the API for etcd management is represented an installation task.

Now we're saying that etcd is not assumed, and we only manage the etcd operator implementation (and any supporting resources, e.g. CRDs) depending on the API configuration. To me this says that etcd CRDs need to come out of the installation path and become managed resources.

@csrwng @enxebre @sjenning does this make sense? I think the hypershift-operator itself would need to manage the etcd-operator CRDs.

That work could be a followup though.

Reply:
Another thing I just thought of: what happens if you have one managed and one unmanaged control plane? We would need to install global CRDs for the managed one... If you're saying that there can't be any managed etcd in _any control plane_ that appears to be in conflict with the idea that this is a per-cluster API?

Reply:
We can separate the concept of managed etcd per cluster vs managing Etcd CRDs at the operator level. We could have a global config/flag bool for `ManageEtcdCRDs` If true, then the hypershift operator creates the CRDs, otherwise it does nothing. That would be orthogonal to the managed/unmanaged etcd per cluster fields. I also agree that it could be done as a follow up.

Reply:
The main reason why that can't happen now is the CRDs are named the same as the ones we used in the existing etcd operator and the version currently defined isn't compatible with what we are using in production (haven't figured out all the differences yet).

Normally CRDs are named uniquely and that's not a problem. The issue here is there's multiple components trying to claim etcdcluster, etcdbackup, etcdrestore

Reply:
>We can separate the concept of managed etcd per cluster vs managing Etcd CRDs at the operator level. We could have a global config/flag bool for ManageEtcdCRDs If true, then the hypershift operator creates the CRDs, otherwise it does nothing.

Not sure I follow this. Why do we need an etcd management API on hostedcluster if we effectively have an installation-level config that imposes requirements on all hosted clusters? What would it mean to have a global "unmanaged etcd" setting with individual clusters configured for managed etcd?

cc @derekwaynecarr

Reply:
It's different things. It's not a global "unmanaged etcd", it's a global "unmanaged etcd crd". I can still try to manage etcd for my cluster even though its CRD was not installed by me.

Reply:
I'm not sure I agree these are different things. The CRDs in question exist only in the context of the built-in/default implementation for "managed etcd". The CRDs/Deployment/SA/Roles for the etcd-operator taken altogether represent the managed etcd implementation. There is no meaningful decoupling.

My view of it remains that if you ask for a managed etcd for your cluster, the operator is responsible for creating and operating an etcd instance for that cluster (including global resources like CRDs), and that's something that can only be interpreted per-cluster and at runtime.

So I'm still convinced there should be _no global etcd config_, and the CRD should be managed by the operator. As to this conflict:

>The main reason why that can't happen now is the CRDs are named the same as the ones we used in the existing etcd operator and the version currently defined isn't compatible with what we are using in production

One side effect of our choice of default implementation is that it relies on globals (CRDs) that preclude multiple co-existent etcd-operator implementations or versions, whether managed by HyperShift or by the user in their management cluster for other reasons. So the answers here are things like:

- Don't create any hostedclusters with managed etcd in a cluster where the user is already self-managing a conflicting component
- Change the etcd-operator so that its CRDs are using an openshift or hypershift API group and so don't present any conflict (this makes more sense to me)

Reply:
Just had a quick convo with @csrwng and I think we agree that we can move forward with the flag here to get us moving here, and what we really need next is more in depth discussion with the etcd team to understand how to incorporate managed etcd into the main openshift API — that is, the fact we're managing a CRD _at all_ is probably a big red flag, because the default managed etcd is supposed to be a supported part of OCP in this context backed up by openshift API, not a bolt-on API like we have today.

Imagine that the etcd operator used for hypershift is part of the OCP payload (which seems desirable/inevitable) — how could its API live external to the payload (i.e. in the operator) at all? That operator would be reconciling OCP API just like other components.

We can take this discussion elsewhere and move forward with the flag as proposed as a stopgap.